### PR TITLE
Instead of using a hard-coded list, pull the channels from altair API

### PR DESCRIPTION
**Pull request by chmduquesne**
In order to avoid that a change in the altair API makes this project outdated, I suggest pulling the channel encoding from altair directly.

### CODE REVIEW

**Comment by chmduquesne** [comment]
Meh. Apparently functools.lru_cache is too new. I'll modify my PR.

**Comment by chmduquesne** [comment]
I squashed my commits into one for the sake of clarity

**Comment by stsievert** [comment]
Thanks for these PRs!

I would like some ordering to this. The `x`, `y` and `color` encodings are most common, and I'd like them to be at the top. I think the rest should be alphabetical though.

For reference, the current ordering with altair 2.2.2 is

``` python
['color',
 'column',
 'detail',
 'fill',
 'href',
 'key',
 'latitude',
 'latitude2',
 'longitude',
 'longitude2',
 'opacity',
 'order',
 'row',
 'shape',
 'size',
 'stroke',
 'text',
 'tooltip',
 'x',
 'x2',
 'y',
 'y2']
```

**Comment by chmduquesne** [comment]
That should do the trick!

File: altair_widgets/widget.py [hunk 1]
```diff
@@ -232,8 +232,18 @@ def _get_types():
...
+    encodings = []
+    for name in dir(altair):
+        value = getattr(altair, name)
+        if (type(value) is type and
+            issubclass(value, altair.FieldChannelMixin) and
+            name != 'FieldChannelMixin'):
+            encodings.append(name.lower())
+    # reorder to have the most useful encodings at the top
+    top = ['x', 'y', 'color']
+    list(map(encodings.remove, top))
```
Comment (stsievert) [inline altair_widgets/widget.py]: It'd nice to have this be a list comp; `[x for x in encodings if x not in top]`.

**Pull request merged**